### DISCUSSION TITLE
No suitable <iframe> found to embed the inspector.

### DISCUSSION THREAD

**Discussion by jorabin** [statelyai/xstate#4106] [Q&A] [answered]
This may be similar to #3701 and it's possible I am being very stupid (apologies if so) but I'm trying to follow the instructions for inspector and I'd like to embed it in an iframe. 

[This is the code/sandbox](https://codesandbox.io/s/polished-fog-mx5lk7?file=/src/index.tsx)

It works in a separate window, (i.e. iframe: false), but when I try to use an iframe I get:

    "No suitable <iframe> found to embed the inspector. Please pass an <iframe> element to `inspect(iframe)` or create an <iframe data-xstate></iframe> element. "

despite there being a 

         <iframe data-xstate title="iframe" width="100%"></iframe>

I'm really a beginner in XState and React and I am wondering if this is because the `inspect()` call is (as instructed) as early as possible, before the iframe is created?

**Comment by Andarist** [comment, marked as answer]
> I'm really a beginner in XState and React and I am wondering if this is because the inspect() call is (as instructed) as early as possible, before the iframe is created?

That's exactly the issue here. `inspect` tries to look for an iframe right away - when you call it - but the iframe is only created by React when you call `root.render(<App />);`.

I'd recommend you to move this `<iframe/>` directly to your HTML page, like here: https://codesandbox.io/s/hopeful-http-hwfylm?file=/src/index.tsx

**Reply by jorabin** [reply]
thanks for this
